vault/request_handling.go

```diff
@@ -2339,6 +2336,7 @@ func (c *Core) LocalUpdateUserFailedLoginInfo(ctx context.Context, userKey Faile
 		// delete the entry from the map, if no key exists it is no-op
 		delete(c.userFailedLoginInfo, userKey)
 	}
+	c.userFailedLoginInfoLock.Unlock()
```

Comment:
Don't we want to keep the defer Unlock? Otherwise we could exit early and never release it...

Reply:
We should. I added that as a fix of part of this change so the revert took that out. I can add that back in as a different PR.